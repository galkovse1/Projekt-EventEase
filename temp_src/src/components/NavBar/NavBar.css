.navbar {
  overflow-x: hidden;
  overflow-y: hidden;
}

.icon-size {
  font-size: 20px;
}
