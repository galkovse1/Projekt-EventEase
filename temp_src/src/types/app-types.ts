import { ActualFileObject } from 'filepond';
import { Socket } from 'socket.io-client';
import { MultiValue } from 'react-select';

export interface IUser {
  id?: number;
  email: string;
  picUrl?: string;
  publicPic_id?: string;
  username?: string;
  name?: string;
  bio?: string;
  lastName?: string;
}

export interface ServerErrors {
  name: string;
  description: string;
  trace: string;
  status: number;
  code: number;
  error: boolean;
}

export interface IUserContext {
  userInfo: IUserAPI | null;
  setUserInfo: React.Dispatch<IUserAPI>;
}

export interface IEvents {
  id: number;
  createdBy: number;
  location: string;
  eventDate: string;
  eventName: string;
  description: string;
  picUrl: string;
  publicPic_id?: string;
  topic: string;
  members?: Array<IMembersAPI>;
  user: IUserAPI;
}

export interface IEventsData {
  createdBy: string;
  location: string;
  eventDate: string;
  eventName: string;
  description: string;
  eventPic: ActualFileObject;
}

export interface IOptionsForMembers {
  value: string;
  label: string;
}

export interface IUserAPI {
  id: number;
  email: string;
  picUrl: string;
  publicPic_id?: string;
  username?: string;
  name: string;
  bio?: string;
  lastName?: string;
}

export interface IDataMembers {
  members: MultiValue<IOptionsForMembers[]>;
}

export interface ICurrentEventContext {
  eventData: IEvents | null;
  updateCurrentEvent: (event: IEvents) => void;
}

export interface IMembersAPI {
  id: number;
  activeMember: boolean;
  event_id: number;
  user_id: number;
}

export interface IDataItems {
  item_name: string;
  id: number;
}

export interface IListItems {
  id: number;
  event_id: number;
  item_name: string;
}

export interface ISingleEvent {
  id: number;
  location: string;
  eventDate: string;
  eventName: string;
  description: string;
  picUrl: string;
  topic: string;
  createdBy: number;
  members: Array<IMembersAPI>;
  items: Array<IListItems>;
  user: IUserAPI;
}

export interface ICollaboratorsList {
  id: number;
  item_id: number;
  added: boolean;
  member_id: number;
  user: IUserAPI;
}

export interface IEditUser {
  email: string;
  name: string;
  bio: string;
  lastName: string;
  username: string;
  userPic: ActualFileObject | string;
}

export interface ISocket {
  socket: Socket | null;
}

export interface IMessages {
  id: number;
  message: string;
  event_id: number;
  user_id: number;
  roomChatId: string;
  createdAt: string;
  user?: IUserAPI;
}

export interface IErrorSockets {
  message: string;
  error: boolean;
  code: number;
}

export interface ICalendarEvent {
  title: string;
  start: Date;
  end: Date;
  allDay?: boolean;
}
