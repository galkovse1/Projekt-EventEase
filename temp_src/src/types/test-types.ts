import { User } from '@auth0/auth0-react';

export interface MockedUserAuth0 extends User {
  mockReturnValue: jest.Mock;
}
