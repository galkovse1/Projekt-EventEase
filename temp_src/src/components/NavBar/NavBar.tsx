import { useState } from 'react';
import { useAuth0 } from '@auth0/auth0-react';
import { Link } from 'react-router-dom';
import { HiMenuAlt3 } from 'react-icons/hi';
import { BsCalendar2Event } from 'react-icons/bs';
import { RiLogoutBoxRLine } from 'react-icons/ri';
import { AiOutlineUser } from 'react-icons/ai';
import { CgAddR } from 'react-icons/cg';

import './NavBar.css';

export const NavBar = () => {
  const { logout } = useAuth0();
  const [open, setOpen] = useState(true);

  return (
    <header className="navbar">
      <div className={`bg-[rgb(62,62,63)] min-h-screen ${open ? 'w-72' : 'w-16'} duration-500 text-gray-100 px-4`}>
        <div className="py-3 flex justify-end">
          <HiMenuAlt3 size={26} className="cursor-pointer" onClick={() => setOpen(!open)} />
        </div>
        <nav className="mt-4 flex flex-col gap-4 relative">
          <Link
            to="/"
            className={`group flex items-center text-sm  gap-3.5 font-medium p-2 hover:bg-slate-500 rounded-md cursor-pointer`}
          >
            <div>
              <BsCalendar2Event className="icon-size" />
            </div>
            <h2 className={`whitespace-pre duration-500 ${!open ? 'opacity-0 translate-x-28 overflow-hidden' : ''}`}>
              Events
            </h2>
          </Link>

          <Link
            to="/profile"
            className={`group flex items-center text-sm  gap-3.5 font-medium p-2 hover:bg-slate-500 rounded-md cursor-pointer`}
          >
            <div>
              <AiOutlineUser className="icon-size" />
            </div>
            <h2 className={`whitespace-pre duration-500 ${!open ? 'opacity-0 translate-x-28 overflow-hidden' : ''}`}>
              Profile
            </h2>
          </Link>
          <Link
            to="/create-event"
            className={`group flex items-center text-sm  gap-3.5 font-medium p-2 hover:bg-slate-500 rounded-md cursor-pointer`}
          >
            <div>
              <CgAddR className="icon-size" />
            </div>
            <h2 className={`whitespace-pre duration-500 ${!open ? 'opacity-0 translate-x-28 overflow-hidden' : ''}`}>
              Create Event
            </h2>
          </Link>
          <button
            data-testid="logoutButton"
            onClick={() => logout({ returnTo: window.location.origin })}
            className={`group flex items-center text-sm  gap-3.5 font-medium p-2 hover:bg-slate-500 rounded-md cursor-pointer`}
          >
            <div>
              <RiLogoutBoxRLine className="icon-size" />
            </div>
            <h2 className={`whitespace-pre duration-500 ${!open ? 'opacity-0 translate-x-28 overflow-hidden' : ''}`}>
              Logout
            </h2>
          </button>
        </nav>
      </div>
    </header>
  );
};
