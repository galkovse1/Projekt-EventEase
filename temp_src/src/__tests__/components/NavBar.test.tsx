import { render, screen, fireEvent } from '@testing-library/react';
import { create } from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom/server';
import { useAuth0 } from '@auth0/auth0-react';

import { NavBar } from '../../components/NavBar/NavBar';
import { MockedUserAuth0 } from '../../types/test-types';

jest.mock('@auth0/auth0-react');

describe('NavBar Component tests', () => {
  test('Component should render information', () => {
    const logout = jest.fn();
    (useAuth0 as unknown as MockedUserAuth0).mockReturnValue({
      logout,
    });

    const treeNavBar = create(
      <StaticRouter location={'/'}>
        <NavBar />
      </StaticRouter>
    ).toJSON();

    expect(treeNavBar).toMatchSnapshot();
  });

  test('Logout Process using Auth0', () => {
    const logout = jest.fn();
    (useAuth0 as unknown as MockedUserAuth0).mockReturnValue({
      logout,
    });

    render(
      <StaticRouter location={'/'}>
        <NavBar />
      </StaticRouter>
    );

    const logoutBtn = screen.getByTestId('logoutButton');
    fireEvent.click(logoutBtn);

    expect(useAuth0).toHaveBeenCalled();
    expect(logout).toHaveBeenCalledTimes(1);
  });
});
