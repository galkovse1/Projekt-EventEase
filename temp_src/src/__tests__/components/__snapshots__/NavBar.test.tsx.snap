// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component tests Component should render information 1`] = `
<header
  className="navbar"
>
  <div
    className="bg-[rgb(62,62,63)] min-h-screen w-72 duration-500 text-gray-100 px-4"
  >
    <div
      className="py-3 flex justify-end"
    >
      <svg
        className="cursor-pointer"
        fill="currentColor"
        height={26}
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 20 20"
        width={26}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM9 15a1 1 0 011-1h6a1 1 0 110 2h-6a1 1 0 01-1-1z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <nav
      className="mt-4 flex flex-col gap-4 relative"
    >
      <a
        className="group flex items-center text-sm  gap-3.5 font-medium p-2 hover:bg-slate-500 rounded-md cursor-pointer"
        href="/"
        onClick={[Function]}
      >
        <div>
          <svg
            className="icon-size"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11 7.5a.5.5 0 0 1 .5-.5h1a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5h-1a.5.5 0 0 1-.5-.5v-1z"
            />
            <path
              d="M3.5 0a.5.5 0 0 1 .5.5V1h8V.5a.5.5 0 0 1 1 0V1h1a2 2 0 0 1 2 2v11a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2V3a2 2 0 0 1 2-2h1V.5a.5.5 0 0 1 .5-.5zM2 2a1 1 0 0 0-1 1v11a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1V3a1 1 0 0 0-1-1H2z"
            />
            <path
              d="M2.5 4a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5H3a.5.5 0 0 1-.5-.5V4z"
            />
          </svg>
        </div>
        <h2
          className="whitespace-pre duration-500 "
        >
          Events
        </h2>
      </a>
      <a
        className="group flex items-center text-sm  gap-3.5 font-medium p-2 hover:bg-slate-500 rounded-md cursor-pointer"
        href="/profile"
        onClick={[Function]}
      >
        <div>
          <svg
            className="icon-size"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 1024 1024"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
            />
          </svg>
        </div>
        <h2
          className="whitespace-pre duration-500 "
        >
          Profile
        </h2>
      </a>
      <a
        className="group flex items-center text-sm  gap-3.5 font-medium p-2 hover:bg-slate-500 rounded-md cursor-pointer"
        href="/create-event"
        onClick={[Function]}
      >
        <div>
          <svg
            className="icon-size"
            fill="none"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 6C12.5523 6 13 6.44772 13 7V11H17C17.5523 11 18 11.4477 18 12C18 12.5523 17.5523 13 17 13H13V17C13 17.5523 12.5523 18 12 18C11.4477 18 11 17.5523 11 17V13H7C6.44772 13 6 12.5523 6 12C6 11.4477 6.44772 11 7 11H11V7C11 6.44772 11.4477 6 12 6Z"
              fill="currentColor"
            />
            <path
              clipRule="evenodd"
              d="M5 22C3.34315 22 2 20.6569 2 19V5C2 3.34315 3.34315 2 5 2H19C20.6569 2 22 3.34315 22 5V19C22 20.6569 20.6569 22 19 22H5ZM4 19C4 19.5523 4.44772 20 5 20H19C19.5523 20 20 19.5523 20 19V5C20 4.44772 19.5523 4 19 4H5C4.44772 4 4 4.44772 4 5V19Z"
              fill="currentColor"
              fillRule="evenodd"
            />
          </svg>
        </div>
        <h2
          className="whitespace-pre duration-500 "
        >
          Create Event
        </h2>
      </a>
      <button
        className="group flex items-center text-sm  gap-3.5 font-medium p-2 hover:bg-slate-500 rounded-md cursor-pointer"
        data-testid="logoutButton"
        onClick={[Function]}
      >
        <div>
          <svg
            className="icon-size"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M5 22a1 1 0 0 1-1-1V3a1 1 0 0 1 1-1h14a1 1 0 0 1 1 1v3h-2V4H6v16h12v-2h2v3a1 1 0 0 1-1 1H5zm13-6v-3h-7v-2h7V8l5 4-5 4z"
              />
            </g>
          </svg>
        </div>
        <h2
          className="whitespace-pre duration-500 "
        >
          Logout
        </h2>
      </button>
    </nav>
  </div>
</header>
`;
